source/octf/node/NodeBase.cpp

```diff
@@ -60,16 +66,21 @@ bool NodeBase::readSettings() {
 
 bool NodeBase::writeSettings() {
     if (m_config) {
-        ProtobufReaderWriter rw(
-                getFrameworkConfiguration().getNodeSettingsFilePath(
-                        getNodePath()));
+        try {
+            ProtobufReaderWriter rw(
+                    getFrameworkConfiguration().getNodeSettingsFilePath(
+                            getNodePath()));
 
-        if (rw.write(*m_config)) {
-            return true;
-        } else {
-            log::cerr << "Cannot write configuration for file "
-                      << rw.getFilePath() << std::endl;
+            if (rw.write(*m_config)) {
```

Comment:
👍 It's good idea to try catch in write path which is involved during de-initialization. Thus allows to perform another de-initialization activities.